custom_components/frigate/views.py

```diff
@@ -208,9 +210,36 @@ def _create_path(self, path: str, **kwargs: Any) -> str | None:
             return f"api/events/{event_id}/clip.mp4"
         return None
 
-    def _permit_request(self, request: web.Request, config_entry: ConfigEntry) -> bool:
+    def _permit_request(self, request: web.Request, config_entry: ConfigEntry, **kwargs: Any) -> bool:
         """Determine whether to permit a request."""
-        return bool(config_entry.options.get(CONF_NOTIFICATION_PROXY_ENABLE, True))
+
+        is_notification_proxy_enabled = bool(config_entry.options.get(CONF_NOTIFICATION_PROXY_ENABLE, True))
+
+        # If proxy is disabled, immediately reject
+        if not is_notification_proxy_enabled:
+            return False
+
+        notification_expiration_mins = int(config_entry.options.get(CONF_EXPIRE_NOTIFICATIONS_AFTER_MINS, 0))
+
+        # If notification events never expire, immediately permit
+        if notification_expiration_mins == 0:
```

Comment:
This is where I was hoping you could check if the request is authenticated/signed, and automatically permit if yes. i.e. only unauthenticated requests are subject to expiration.

Reply:
Sure, just check the signature like we do in VodSegmentProxyView? I am assuming we'd share that logic in a common method. Any preferences for where that helper goes?

Reply:
I think there might be an easier path for you. The auth middleware may be automatically adding a key to the request if the request is authenticated, see: https://github.com/home-assistant/core/blob/dev/homeassistant/components/http/auth.py#L137

So just check if `request[KEY_AUTHENTICATED]` is a value you can just use? [Have not tested this]